src/components/shared/pcsGrid/pcsGrid.js

```diff
@@ -62,6 +65,22 @@ export class PcsGrid extends Component {
     }
   }
 
+  /** Ensure that hard selected rows are maintained by their ids, even when the actual data may change */
+  componentDidUpdate(nextProps, nextState) {
+    const { idName = 'id' } = this.props; //Allow override of the name of the identifier property on the data items.
```

Comment:
The pcsGrid already has a prop that knows how to access an id called `getSoftSelectId`. If you want to allow this functionality use that.